src/main/scala/is/hail/io/annotators/IntervalList.scala

```diff
@@ -44,23 +46,33 @@ object IntervalList {
 
     val rdd = hc.sc.textFileLines(filename)
       .filter(l => !l.value.isEmpty && l.value(0) != '@')
-      .map {
+      .flatMap {
         _.map { line =>
           if (hasValue) {
             val split = line.split("\\s+")
             split match {
               case Array(contig, start, end, dir, target) =>
-                val interval = Interval(Locus.annotation(contig, start.toInt, rg), Locus.annotation(contig, end.toInt, rg),
-                  includesStart = true, includesEnd = true)
-                Row(interval, target)
+                if (skipInvalidIntervals && rg.forall(rg => !rg.isValidContig(contig) || !rg.isValidLocus(contig, start.toInt) || !rg.isValidLocus(contig, end.toInt))) {
+                  warn(s"Interval is not consistent with reference genome '${ rg.get.name }': '$line'.")
+                  None
+                } else {
+                  val interval = Interval(Locus.annotation(contig, start.toInt, rg), Locus.annotation(contig, end.toInt, rg),
+                    includesStart = true, includesEnd = true)
+                  Some(Row(interval, target))
+                }
               case arr => fatal(s"expected 5 fields, but found ${ arr.length }")
             }
           } else {
             line match {
               case intervalRegex(contig, start, end) =>
-                val interval = Interval(Locus.annotation(contig, start.toInt, rg), Locus.annotation(contig, end.toInt, rg),
-                  includesStart = true, includesEnd = true)
-                Row(interval)
+                if (skipInvalidIntervals && rg.forall(rg => !rg.isValidContig(contig) || !rg.isValidLocus(contig, start.toInt) || !rg.isValidLocus(contig, end.toInt))) {
```

Comment:
exists

Reply:
I think this behavior will produce no intervals if `skip_invalid_intervals` is True and rg is None. Can you write a test to make sure this doesn't happen?